### PR TITLE
Add Preset data for department and designation for initial installation

### PR DESCRIPTION
**Pull request by ManiruzzamanAkash**
[update] Added Preset data for department and designation for first time installation

### CODE REVIEW

File: includes/class-install.php [hunk 1]
```diff
@@ -746,6 +746,8 @@ public function create_tables() {
                 `school` varchar(100) DEFAULT NULL,
                 `degree` varchar(100) DEFAULT NULL,
                 `field` varchar(100) DEFAULT NULL,
+                `result` decimal(4,3) DEFAULT 0,
+                `result_type` enum('gpa', 'cgpa') DEFAULT NULL,
```
Comment (tanmayjay) [inline includes/class-install.php:750]: Kindly check these changes exist in #1242 with different definition!

File: includes/updates/update-1.8.3.php [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+            $wpdb->prepare(
+                "ALTER TABLE `{$wpdb->prefix}erp_hr_education` ADD `result` decimal(4,3) NULL DEFAULT '0' AFTER `field`;"
+            )
+        );
+    }
+
+    if ( ! in_array( 'result_type', $cols ) ) {
+        $wpdb->query(
+            $wpdb->prepare(
+                "ALTER TABLE `{$wpdb->prefix}erp_hr_education` ADD `result_type` ENUM('cgpa','gpa') NULL DEFAULT NULL AFTER `result`;"
```
Comment (tanmayjay) [inline includes/updates/update-1.8.3.php:22]: Kindly check these changes exist in #1242 with different definition!

File: modules/hrm/includes/functions-employee.php [hunk 3]
```diff
@@ -1105,3 +1105,27 @@ function get_employee_additional_fields( $fields, $id, $user ) {
...
+    ];
+
+    $types = apply_filters( 'erp_hr_education_result_type_option', $types );
+
+    if ( $selected ) {
+        return ( isset( $types[ $selected ] ) ) ? $types[ $selected ] : '';
+    }
+
+    return $types;
+}
```
Comment (tanmayjay) [inline modules/hrm/includes/functions-employee.php:1131]:
```
function erp_hr_get_education_result_type_options( $selected = null ) {
    $types = [
        'grade'      => __( 'Grade',  'erp' ),
        'percentage' => __( 'Pecentage', 'erp' )
    ];

    $types = apply_filters( 'erp_hr_education_result_type_option', $types );

    if ( $selected ) {
        return ( isset( $types[ $selected ] ) ) ? $types[ $selected ] : '';
    }

    return $types;
}
```

Same function in #1242 with completely different definition

File: modules/hrm/includes/models/education.php [hunk 4]
```diff
@@ -10,5 +10,5 @@
 class Education extends Model {
     protected $table = 'erp_hr_education';
 
-    protected $fillable = [ 'employee_id', 'school', 'degree', 'field', 'finished', 'notes', 'interest' ];
+    protected $fillable = [ 'employee_id', 'school', 'degree', 'field', 'result', 'result_type', 'finished', 'notes', 'interest' ];
```
Comment (tanmayjay) [inline modules/hrm/includes/models/education.php:13]: Changes from #1242

File: modules/hrm/views/js-templates/education-form.php [hunk 5]
```diff
@@ -46,6 +46,33 @@
...
+            'label'    => __( 'Result in', 'erp' ),
+            'name'     => 'result_type',
+            'value'    => '{{ data.result_type }}',
+            'required' => true,
+            'type'     => 'select',
+            'id'       => 'result_type',
+            'options'  => [ '' => __( '- Select -', 'erp' ) ] + erp_hr_get_education_result_type_options(),
+        ] ); ?>
+    </div>
+
```
Comment (tanmayjay) [inline modules/hrm/views/js-templates/education-form.php:75]: Same changes from #1242  with different codes!

**Review by tanmayjay — CHANGES_REQUESTED** [review]

**Comment by ManiruzzamanAkash** [comment]
Fixed redundant code in different PR.

**Review by tanmayjay — APPROVED** [review]

**Pull request merged**